packages/blob/src/client.ts

```diff
@@ -232,8 +236,12 @@ export async function handleUpload({
   const type = body.type;
   switch (type) {
     case 'blob.generate-client-token': {
-      const { pathname, callbackUrl } = body.payload;
-      const payload = await onBeforeGenerateToken(pathname);
+      const { pathname, callbackUrl, clientPayload } = body.payload;
+      const payload = await onBeforeGenerateToken(pathname, clientPayload);
+      // TODO @Fabio: the next line `?? null` previously, which always made sure that tokenPayload was `null`
+      // Why so?
+      const tokenPayload = payload.tokenPayload ?? clientPayload;
```

Comment:
that's a bit confusing indeed...we can generate a payload from the server (through `onBeforeGenerateToken`, think the case we need to send the session Id) or from the client (directly through the `clientPayload` for any unsecure data). Which of the two values should have precedence? should we replace this logic by a function, which potentially alows to combine server and client payload?

Reply:
oh I see now, we are passing the `clientPayload` to the `onBeforeGenerateToken`. That should be good then (not sure what's the reason of your comment then).

May not be clear for the user that  `onBeforeGenerateToken` can return a tokenPayload...maybe an optional async `onGeneratePayload(clientToken)` would be more clear?

Reply:
Discussed in call: we keep it this way. The naming internally is not obvious between the various uses of the word "payload" but that's on us.